Popup menu order

DollKM opened the pull request:
## Submission Description
### Feature Addition - Adding an "order" Attribute to PopupMenu Options
**Overview:**
This submission aims to add a new attribute called "order" to the PopupMenu::Item class, enabling the sorting of these options.

**Proposal:**
To ensure the relevance and effectiveness of this new feature, I provide the following proposal to explain the purpose and use of this feature:

- **Purpose:**
The addition of the "order" attribute will allow developers to arrange PopupMenu options in a specific order. This is particularly useful for cases where custom ordering of options is required, such as sorting options based on business needs.
- **Use Case:**
Developers can specify the display order of each option by setting the "order" attribute. Smaller "order" values will appear below the menu, while larger values will appear above it. This provides users with greater flexibility in organizing and presenting PopupMenu options.
### Modifications Made
- Added a new attribute called "order" to the PopupMenu::Item class.
- Added a set_item_order function to the PopupMenu class.
Implemented sorting of items in descending order based on the "order" attribute when the menu is opened.
### Usage Example
```GDScript
var id = 99
scene_menu.add_item('test', id)
scene_menu.set_item_order(id, 100)
```
### Considerations
It is recommended to use "order" values >= 100, but if you prefer to use a smaller number, it is also acceptable.
If multiple options have the same "order" value, they will be displayed in the order they were created in the code.

YuriSizov commented:
Please make separate pull requests for separate features. New features also need to come with a proposal, so their relevance can be discussed and evaluated (proposals should be one per feature as well).

KoBeWi commented:
1. There is already `about_to_popup` signal, which does the same from what I understand.

DollKM commented:
> 1. There is already `about_to_popup` signal, which does the same from what I understand.

Thank you for your suggestion; I hadn't noticed this signal before.
I have removed the 'menu_opened' signal in the new commit.

YeldhamDev commented:
You should probably also checkout the docs about PR workflow (mainly about updating one's PR):
https://docs.godotengine.org/en/stable/contributing/workflow/pr_workflow.html#modifying-a-pull-request

DollKM commented:
I'm sorry, this is my first time contributing code to an open-source project, and I didn't read the documentation carefully. Now it's okay. I have organized my commit and completed the proposal.

AThousandShips reviewed (commented):
Required for this to pass CI

AThousandShips commented on scene/gui/popup_menu.cpp:
```diff
@@ -2162,6 +2195,7 @@ void PopupMenu::_get_property_list(List<PropertyInfo> *p_list) const {
 
 void PopupMenu::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("activate_item_by_event", "event", "for_global_only"), &PopupMenu::activate_item_by_event, DEFVAL(false));
+	ClassDB::bind_method(D_METHOD("set_item_order", "id", "order"), &PopupMenu::set_item_order, DEFVAL(-1), DEFVAL(0));
```
```suggestion
	ClassDB::bind_method(D_METHOD("set_item_order", "id", "order"), &PopupMenu::set_item_order);
```
These default values doesn't make sense, if `id` is `-1` it won't affect any item, and order should be provided if you're setting it no?

DollKM committed:
[Feature] Add "order" attribute to PopupMenu::Item

- Added a new "order" attribute to the `PopupMenu::Item` class.
- Introduced a `set_item_order` function in the `PopupMenu` class.
- Implemented descending sorting of items based on the "order" attribute when the menu is opened.
This enhancement allows developers to customize the display order of PopupMenu options, providing greater flexibility in organizing and presenting menu items.

Usage Example:
```GDScript
var id = 99
scene_menu.add_item('test', id)
scene_menu.set_item_order(id, 100)

AThousandShips commented:
> completed the proposal.

You need to open a proposal, not write in your PR description
